src/passes/OptimizeInstructions.cpp

```diff
@@ -458,6 +458,90 @@ struct OptimizeInstructions
           return c;
         }
       }
+      {
+        Const* c;
+        Binary* bin;
+        Expression *x, *y;
+        BinaryOp op;
+
+        // fneg(x) * fneg(y)   ==>   x * y
+        // fneg(x) / fneg(y)   ==>   x / y
+        if (matches(
+              curr,
+              binary(&bin, Mul, unary(Neg, any(&x)), unary(Neg, any(&y)))) ||
+            matches(
+              curr,
+              binary(&bin, DivS, unary(Neg, any(&x)), unary(Neg, any(&y))))) {
+          bin->left = x;
+          bin->right = y;
+          return bin;
+        }
+        // fneg(x * C)   ==>   x * -C
+        // fneg(x / C)   ==>   x / -C
+        // fneg(C / x)   ==>   -C / x
+        //  if x != C and C != NaN
```

Comment:
why does the comment say `x != C`?

Reply:
because if `x` is constant it could be NaN. So for fast reject this we just skip all constants. Anyway case when both are constant is better handle in constant propagation pass

Reply:
Ah, that's `fastMath` case, so we could simplify this as well